### DISCUSSION TITLE
OpenEVSE Lifetime kWh source

### DISCUSSION THREAD

**Discussion by fhteagle** [firstof9/openevse#169] [Q&A] [answered]
Trying to help @KipK troubleshoot a display bug in the new GUI. HA displays the lifetime usage correctly in sensor.<unitname>_total_usage, but the new GUI sometimes does not. How does HA pull the lifetime kWh from the unit itself? RAPI, or a specific HTTP API endpoint?

**Comment by firstof9** [comment, marked as answer]
The config and status endpoints initially, then websocket for updates.

**Comment by fhteagle** [comment]
Okay issue was found and fixed in the OpenEVSE Wifi firmware after all. Thanks.
